### PR TITLE
DO NOT MERGE: Alternative proposal for go/firestore-snapshot-reads

### PR DESCRIPTION
**Pull request by dconeybe**

### CODE REVIEW

File: firebase-firestore/src/main/java/com/google/firebase/firestore/ReadOnlyTransaction.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  @NonNull
+  public DocumentSnapshot get(@NonNull DocumentReference documentRef)
+      throws FirebaseFirestoreException {
+    return get(documentRef, null);
+  }
+
+  @NonNull
+  public DocumentSnapshot get(@NonNull DocumentReference documentRef, @Nullable Timestamp readTime)
```
Comment (schmidt-sebastian) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/ReadOnlyTransaction.java:34]: FWIW While we could implement this, this breaks the transaction concept of using the same read time across all reads.

Comment (dconeybe) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/ReadOnlyTransaction.java:34]: Okay, I'll fix this.

Comment (schmidt-sebastian) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/ReadOnlyTransaction.java:34]: Thanks. We need to ensure that a transaction always returns a consistent view of data.

Comment (dconeybe) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/ReadOnlyTransaction.java]: Done.

**Comment by dconeybe** [comment]
This feature has been put on hold indefinitely

**Pull request closed without merging**